Foundation/NSEnergyFormatter.swift

```diff
@@ -7,41 +7,195 @@
 // See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
 //
 
-
 extension EnergyFormatter {
-    public enum Unit : Int {
+    public enum Unit: Int {
+
+        case joule = 11
+        case kilojoule = 14
+        case calorie = 1793 // chemistry "calories", abbr "cal"
+        case kilocalorie = 1794 // kilocalories in general, abbr “kcal”, or “C” in some locales (e.g. US) when usesFoodEnergy is set to YES
+
+        // Map Unit to UnitEnergy class to aid with conversions
+        fileprivate var unitEnergy: UnitEnergy {
+            switch self {
+            case .joule:
+                return UnitEnergy.joules
+            case .kilojoule:
+                return UnitEnergy.kilojoules
+            case .calorie:
+                return UnitEnergy.calories
+            case .kilocalorie:
+                return UnitEnergy.kilocalories
+            }
+        }
+
+        // Reuse symbols defined in UnitEnergy, except for kilocalories, which is defined as "kCal"
+        fileprivate var symbol: String {
+            switch self {
+            case .kilocalorie:
+                return "kcal"
+            default:
+                return unitEnergy.symbol
+            }
+        }
+
+        // Return singular, full string representation of the energy unit
+        fileprivate var singularString: String {
+            switch self {
+            case .joule:
+                return "joule"
```

Comment:
This is missing the meat of what `EnergyFormatter` provides on Apple platforms: namely, formatting energy values in a locale-appropriate way. On macOS:

```swift
let formatter = EnergyFormatter()
formatter.numberFormatter.locale = Locale(identifier: "de_DE")
formatter.unitStyle = .long
formatter.string(fromValue: 0, unit: .joule) // "0 Joule"
formatter.numberFormatter.locale = Locale(identifier: "fr_FR")
formatter.string(fromValue: 0, unit: .joule) // "0 joule"
formatter.numberFormatter.locale = Locale(identifier: "zh_CN")
formatter.string(fromValue: 0, unit: .joule) // "0焦耳"
```